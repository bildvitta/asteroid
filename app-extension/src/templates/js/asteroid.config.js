module.exports = {
  /**
   * Configurações de API
   * @type {{ serverTimeout: number=10000 }}
  */
  api: {
    serverTimeout: 10000
  },

  framework: {
    /**
     * O asteroid ao identificar que a aplicação tenha dependências terceiras e
     * não estão declarados na opção "thirdPartyComponents" ele remove as dependências da aplicação (package.json), caso precise dessas dependências mesmo que sem os thirdPartyComponents, desative esta opção setando como "false".
     * @type {boolean=true}
    */
    autoRemoveThirdDependencies: true,

    fonts: {
      observer: {
        /**
         * O asteroid adiciona um observer nas fontes, enquanto ela não carrega adiciona um loading para
         * não quebrar o layout, a configuração "waitForUserAuthenticate: true" define que o observer deve esperar
         * o usuário estar autenticado para começar a observar as fontes, para isto a aplicação deve poder receber um
         * "postMessage" com um evento type "requestUser" e responder com um evento type "responseUser".
         *
         * @type {boolean=true}
         * @example
         * // solicita o usuário
         * window.postMessage({ type: 'requestUser' })
         *
         * window.addEventListener('message', ({ data }) => data.type // responseUser })
       */
        waitForUserAuthenticate: true
      }
    },

    /**
     * Controla o sistema de notificações da aplicação, ao ativado e configurado,
     * estará disponível os seguintes recursos:
     *
     * - Item de notificação no componente QasAppUser;
     * - Toast de notificação em real time;
     * - Ícone de notificação no menu em real time.
     *
     * @type {{ useNotifications: boolean }}
     */
    featureToggle: {
      useNotifications: false
    },

    /**
     * Componentes que necessitam de instalação de bibliotecas de terceiros
     * dentro da aplicação que utiliza o asteroid, a instalação é feita automaticamente pelo asteroid.
     * @type {Array<'QasChartView' | 'QasMap'>}
    */
    thirdPartyComponents: []
  }
}
