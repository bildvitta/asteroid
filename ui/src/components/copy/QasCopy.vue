<template>
  <span>
    <span data-table-hover>
      <slot v-if="props.useText">
        {{ props.text }}
      </slot>
    </span>

    <qas-btn class="q-ml-xs" color="primary" :icon="props.icon" :loading="isLoading" variant="tertiary" @click.stop.prevent="copy">
      <q-tooltip>Copiar</q-tooltip>
    </qas-btn>
  </span>
</template>

<script setup>
import { copyToClipboard } from '../../helpers'
import { ref } from 'vue'

defineOptions({ name: 'QasCopy' })

const props = defineProps({
  icon: {
    default: 'sym_r_file_copy',
    type: String
  },

  text: {
    required: true,
    type: String
  },

  useText: {
    type: Boolean,
    default: true
  },

  rawText: {
    default: '',
    type: String
  }
})

const isLoading = ref(false)

function copy () {
  copyToClipboard(props.rawText || props.text, value => {
    isLoading.value = value
  })
}
</script>
