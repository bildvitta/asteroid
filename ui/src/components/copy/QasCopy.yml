type: component

meta:
  desc: Copia algum texto para o clipboard após clicar no botão.

props:
  icon:
    desc: Ícone para mostrar no botão.
    type: String
    default: sym_r_file_copy

  text:
    desc: Texto a ser copiado.
    type: String

  use-text:
    desc: Controla se o texto vai ser exibido ou não no botão.
    type: Boolean
    default: true

