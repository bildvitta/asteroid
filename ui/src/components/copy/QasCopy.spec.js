import { mount } from '@vue/test-utils'

import '@/index.js'
import QasCopy from './QasCopy.vue'

describe('Test QasCopy component', () => {
  const spyCopyToClipboard = jest.spyOn(QasCopy.methods, 'copy')

  const wrapper = mount(QasCopy, {
    propsData: {
      label: 'Click here to copy!'
    }
  })

  it('test defaults', () => {
    expect(wrapper.element).toMatchSnapshot()

    const buttonElement = wrapper.findComponent({ name: 'QBtn' })
    buttonElement.trigger('click')

    expect(spyCopyToClipboard).toHaveBeenCalled()
    expect(spyCopyToClipboard).toHaveBeenCalledTimes(1)
  })
})
