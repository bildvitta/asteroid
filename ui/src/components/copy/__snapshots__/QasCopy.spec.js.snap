// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test QasCopy component test defaults 1`] = `
<span>
  Click here to copy! 
  <button
    class="q-btn q-btn-item non-selectable no-outline q-ml-xs q-btn--flat q-btn--round text-grey-5 q-btn--actionable q-focusable q-hoverable q-btn--wrap"
    role="button"
    style="font-size: 8px;"
    tabindex="0"
    type="button"
  >
    <span
      class="q-focus-helper"
    />
    <span
      class="q-btn__wrapper col row q-anchor--skip"
    >
      <span
        class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
      >
        <i
          aria-hidden="true"
          class="material-icons-outlined q-icon notranslate"
          role="img"
        >
          file_copy
        </i>
        <!---->
      </span>
    </span>
  </button>
</span>
`;
